### PR TITLE
RM-7662: Remove jQuery usages from BocBooleanValue

### PR DESCRIPTION
**Pull request by Stefan-Ilic**
https://re-motion.atlassian.net/browse/RM-7662

### CODE REVIEW

File: Remotion/ObjectBinding/Web.UnitTests/UI/Controls/BocBooleanValueImplementation/Rendering/BocBooleanValueRendererTest.cs [hunk 1]
```diff
@@ -458,9 +458,9 @@ private void CheckDataValueField (XmlNode outerSpan, string value)
 
     private string GetClickScript (bool isRequired, bool isAutoPostbackEnabled)
     {
-      return "BocBooleanValue.SelectNextCheckboxValue ('ResourceKey', $(this).parent().children('a')[0], "
-             + "$(this).parent().children('a').children('img')[0], $(this).parent().children('span')[1], "
-             + "$(this).parent().children('input')[0], " + isRequired.ToString().ToLower() + ", "
+      return "BocBooleanValue.SelectNextCheckboxValue ('ResourceKey', this.parentNode.querySelector(':scope>a'), "
+             + "this.parentNode.querySelector(':scope>a').querySelector(':scope>img'), this.parentNode.querySelectorAll('span')[1], "
```
Comment (a-ctor) [inline Remotion/ObjectBinding/Web.UnitTests/UI/Controls/BocBooleanValueImplementation/Rendering/BocBooleanValueRendererTest.cs:462]: We can convert this into a single CSS selector

```suggestion
             + "this.parentNode.querySelector(':scope>a>img'), this.parentNode.querySelectorAll('span')[1], "
```

File: Remotion/ObjectBinding/Web/UI/Controls/BocBooleanValueImplementation/Rendering/BocBooleanValueRenderer.cs [hunk 2]
```diff
@@ -253,16 +253,16 @@ private void RegisterStarupScriptIfNeeded (BocBooleanValueRenderingContext rende
       scriptBuilder.Append ("BocBooleanValue.SelectNextCheckboxValue (");
       scriptBuilder.Append ("'").Append (resourceSet.ResourceKey).Append ("'");
       scriptBuilder.Append (", ");
-      scriptBuilder.Append ("$(this).parent().children('a')[0]");
+      scriptBuilder.Append ("this.parentNode.querySelector(':scope>a')");
       scriptBuilder.Append (", ");
-      scriptBuilder.Append ("$(this).parent().children('a').children('img')[0]");
+      scriptBuilder.Append ("this.parentNode.querySelector(':scope>a').querySelector(':scope>img')");
```
Comment (a-ctor) [inline Remotion/ObjectBinding/Web/UI/Controls/BocBooleanValueImplementation/Rendering/BocBooleanValueRenderer.cs:258]: Same here

```suggestion
      scriptBuilder.Append ("this.parentNode.querySelector(':scope>a>img')");
```

File: Remotion/ObjectBinding/Web/UI/Controls/BocBooleanValueImplementation/Rendering/BocBooleanValueRenderer.cs [hunk 3]
```diff
@@ -253,16 +253,16 @@ private void RegisterStarupScriptIfNeeded (BocBooleanValueRenderingContext rende
...
       scriptBuilder.Append (", ");
       if (renderingContext.Control.ShowDescription)
-        scriptBuilder.Append ("$(this).parent().children('span')[1]");
+        scriptBuilder.Append ("this.parentNode.querySelectorAll('span')[1]");
       else
         scriptBuilder.Append ("null");
       scriptBuilder.Append (", ");
-      scriptBuilder.Append ("$(this).parent().children('input')[0]");
+      scriptBuilder.Append ("this.parentNode.querySelector('input')");
```
[the remark this answers is not in the document]
Comment (Stefan-Ilic) [reply Remotion/ObjectBinding/Web/UI/Controls/BocBooleanValueImplementation/Rendering/BocBooleanValueRenderer.cs:265]: Oh, you're right! It's a shame the tests all still work, though 😂😅

**Review by a-ctor — APPROVED** [review]

File: Remotion/ObjectBinding/Web.UnitTests/UI/Controls/BocBooleanValueImplementation/Rendering/BocBooleanValueRendererTest.cs [hunk 4]
```diff
@@ -458,12 +458,12 @@ private void CheckDataValueField (XmlNode outerSpan, string value)
 
     private string GetClickScript (bool isRequired, bool isAutoPostbackEnabled)
     {
-      return "BocBooleanValue.SelectNextCheckboxValue ('ResourceKey', $(this).parent().children('a')[0], "
-             + "$(this).parent().children('a').children('img')[0], $(this).parent().children('span')[1], "
-             + "$(this).parent().children('input')[0], " + isRequired.ToString().ToLower() + ", "
+      return "BocBooleanValue.SelectNextCheckboxValue ('ResourceKey', this.parentNode.querySelector(':scope>a'), "
+             + "this.parentNode.querySelector(':scope>a>img'), this.parentNode.querySelectorAll(':scope>span')[1], "
+             + "this.parentNode.querySelector(':scope>input'), " + isRequired.ToString().ToLower() + ", "
```
Comment (MichaelKetting) [inline Remotion/ObjectBinding/Web.UnitTests/UI/Controls/BocBooleanValueImplementation/Rendering/BocBooleanValueRendererTest.cs:463]:
```suggestion
             + "this.parentNode.querySelector(':scope > a > img'), this.parentNode.querySelectorAll(':scope>span')[1], "
             + "this.parentNode.querySelector(':scope > input'), " + isRequired.ToString().ToLower() + ", "
```

File: Remotion/ObjectBinding/Web/UI/Controls/BocBooleanValueImplementation/Rendering/BocBooleanValueRenderer.cs [hunk 5]
```diff
@@ -253,16 +253,16 @@ private void RegisterStarupScriptIfNeeded (BocBooleanValueRenderingContext rende
       scriptBuilder.Append ("BocBooleanValue.SelectNextCheckboxValue (");
       scriptBuilder.Append ("'").Append (resourceSet.ResourceKey).Append ("'");
       scriptBuilder.Append (", ");
-      scriptBuilder.Append ("$(this).parent().children('a')[0]");
+      scriptBuilder.Append ("this.parentNode.querySelector(':scope>a')");
```
Comment (MichaelKetting) [inline Remotion/ObjectBinding/Web/UI/Controls/BocBooleanValueImplementation/Rendering/BocBooleanValueRenderer.cs:256]:
```suggestion
      scriptBuilder.Append ("this.parentNode.querySelector(':scope > a')");
```

File: Remotion/ObjectBinding/Web/UI/Controls/BocBooleanValueImplementation/Rendering/BocBooleanValueRenderer.cs [hunk 6]
```diff
@@ -253,16 +253,16 @@ private void RegisterStarupScriptIfNeeded (BocBooleanValueRenderingContext rende
...
       scriptBuilder.Append (", ");
-      scriptBuilder.Append ("$(this).parent().children('a').children('img')[0]");
+      scriptBuilder.Append ("this.parentNode.querySelector(':scope>a>img')");
```
Comment (MichaelKetting) [inline Remotion/ObjectBinding/Web/UI/Controls/BocBooleanValueImplementation/Rendering/BocBooleanValueRenderer.cs:258]:
```suggestion
      scriptBuilder.Append ("this.parentNode.querySelector(':scope > a > img')");
```

File: Remotion/ObjectBinding/Web/UI/Controls/BocBooleanValueImplementation/Rendering/BocBooleanValueRenderer.cs [hunk 7]
```diff
@@ -253,16 +253,16 @@ private void RegisterStarupScriptIfNeeded (BocBooleanValueRenderingContext rende
...
       scriptBuilder.Append (", ");
       if (renderingContext.Control.ShowDescription)
-        scriptBuilder.Append ("$(this).parent().children('span')[1]");
+        scriptBuilder.Append ("this.parentNode.querySelectorAll(':scope>span')[1]");
```
Comment (MichaelKetting) [inline Remotion/ObjectBinding/Web/UI/Controls/BocBooleanValueImplementation/Rendering/BocBooleanValueRenderer.cs:261]:
```suggestion
        scriptBuilder.Append ("this.parentNode.querySelectorAll(':scope > span')[1]");
```

File: Remotion/ObjectBinding/Web/UI/Controls/BocBooleanValueImplementation/Rendering/BocBooleanValueRenderer.cs [hunk 8]
```diff
@@ -253,16 +253,16 @@ private void RegisterStarupScriptIfNeeded (BocBooleanValueRenderingContext rende
...
       else
         scriptBuilder.Append ("null");
       scriptBuilder.Append (", ");
-      scriptBuilder.Append ("$(this).parent().children('input')[0]");
+      scriptBuilder.Append ("this.parentNode.querySelector(':scope>input')");
```
Comment (MichaelKetting) [inline Remotion/ObjectBinding/Web/UI/Controls/BocBooleanValueImplementation/Rendering/BocBooleanValueRenderer.cs:265]:
```suggestion
      scriptBuilder.Append ("this.parentNode.querySelector(':scope > input')");
```

**Review by MichaelKetting — CHANGES_REQUESTED** [review]

**Review by MichaelKetting — APPROVED** [review]

**Pull request merged**